diagnostics/include/diagnostics_server.h

```diff
@@ -62,16 +62,16 @@ inline std::string readline(int sock) {
     }
     if (rv < 0 && errno == EINTR) continue;
     if (rv < 0) {
-      throw std::runtime_error("diagnostics server readline select failed: " + errnoString(rv));
+      throw std::runtime_error("diagnostics server readline select failed: " + errnoString(errno));
```

Comment:
I'm not sure if it has any difference but why here and in line 74 we use '+' between the two parts and not '<<'?

Reply:
It's because in the other lines we are using the logging API which uses `<<`. Here and in line 74 we are building a string through concatenation. We aren't "streaming".